main.go

```diff
@@ -275,11 +277,26 @@ func init() {
 
 	cmd = kingpin.MustParse(cli.Parse(os.Args[1:]))
 
+	// Configure logging.
 	switch {
 	case *trace:
 		log.SetLevel(5)
 	case *debug:
 		log.SetLevel(2)
+	default:
+		l := int8(*logLevel)
+		if l < -1 || l > 5 {
+			fmt.Fprintf(os.Stderr, "invalid log level: %d\n", *logLevel)
+			os.Exit(1)
+		}
+
+		if l == -1 {
+			// Zap uses "5" as the value for fatal.
+			// We need to pass in "-5" because `SetLevel` passes the negation.
+			log.SetLevel(-5)
```

Comment:
This technically works because there's no logs with this level. Perhaps there's a better way to disable logging.

Reply:
Yeah, this is unfortunate if it's the best we can do. If it is, I think it's worth leaving an apologetic, explanatory comment.

Reply:
I'm not familiar enough with the logging setup. How would you implement `logr.Discard`?

Reply:
I think this PR provides a cleaner interface to logging and is in a mergable state. I'm unfamiliar with `logr.Discard` as well. We can always follow-up if it turns out this change needs more.